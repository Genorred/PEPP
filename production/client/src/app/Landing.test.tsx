import { render } from '@testing-library/react'
import Home from "@/app/page";

it('Landing is stable', () => {
  const { container } = render(<Home />)
  expect(container).toMatchSnapshot()
})