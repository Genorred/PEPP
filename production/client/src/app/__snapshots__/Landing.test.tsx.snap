// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing is stable 1`] = `
<div>
  <div
    class="page flex min-h-screen flex-col"
  >
    <section
      class="section relative overflow-clip py-20 md:py-32"
    >
      <div
        class="container relative z-10 mx-auto px-4 text-center sm:px-6 lg:px-8 header"
      >
        <h1
          class="font-extrabold tracking-tight"
        >
          Share your 
          <span
            class="text-primary"
          >
            complete
          </span>
           perspective
        </h1>
        <p
          class="mx-auto mt-6 max-w-2xl text-lg text-muted-foreground sm:text-xl"
        >
          Express yourself fully with our wide-angle editor. Connect with friends, get personalized recommendations, and engage in meaningful discussions.
        </p>
        <div
          class="mt-10 flex flex-col items-center justify-center gap-4 sm:flex-row"
        >
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-primary text-primary-foreground hover:bg-primary/90 h-11 rounded-md px-8 gap-2"
          >
            Get Started 
            <svg
              class="lucide lucide-arrow-right h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="m12 5 7 7-7 7"
              />
            </svg>
          </button>
          <button
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 border border-input bg-background hover:bg-accent hover:text-accent-foreground h-11 rounded-md px-8"
          >
            Learn More
          </button>
        </div>
      </div>
      <div
        class="absolute inset-0 -z-10 bg-[radial-gradient(ellipse_at_center,rgba(var(--primary-rgb),0.15),transparent_50%)]"
      />
      <div
        class="mt-8 flex justify-center px-6 sm:mt-20 lg:mt-24"
      >
        <div
          class="relative max-w-3xl rounded-2xl border bg-background shadow-xl"
        >
          <video
            autoplay=""
            class="rounded-2xl"
            height="1200"
            loop=""
            playsinline=""
            width="600"
          >
            <source
              src="/assets/Landing_1.mp4"
              type="video/mp4"
            />
            App interface preview
          </video>
        </div>
      </div>
    </section>
    <section
      class="max-w-full overflow-clip section py-20 bg-muted/50"
      id="features"
    >
      <div
        class="container"
      >
        <div
          class="text-center mb-16"
        >
          <h2
            class="text-3xl font-bold tracking-tight sm:text-4xl md:text-5xl"
          >
            Everything you need to connect
          </h2>
          <p
            class="mx-auto mt-4 max-w-2xl text-muted-foreground"
          >
            Our platform provides all the tools you need to express yourself, connect with others, and discover content that matters to you.
          </p>
        </div>
        <div
          class="grid gap-8 md:grid-cols-2 lg:grid-cols-4 horizontal-move-from-left"
        >
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="mb-4 flex h-12 w-12 items-center justify-center rounded-full bg-primary/10"
            >
              <svg
                class="lucide lucide-square-pen h-6 w-6 text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 3H5a2 2 0 0 0-2 2v14a2 2 0 0 0 2 2h14a2 2 0 0 0 2-2v-7"
                />
                <path
                  d="M18.375 2.625a1 1 0 0 1 3 3l-9.013 9.014a2 2 0 0 1-.853.505l-2.873.84a.5.5 0 0 1-.62-.62l.84-2.873a2 2 0 0 1 .506-.852z"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold"
            >
              Wide-Angle Editor
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Create detailed posts with our expansive editor. Add images, videos, and formatted text to express your complete perspective.
            </p>
          </div>
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="mb-4 flex h-12 w-12 items-center justify-center rounded-full bg-primary/10"
            >
              <svg
                class="lucide lucide-users h-6 w-6 text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M16 21v-2a4 4 0 0 0-4-4H6a4 4 0 0 0-4 4v2"
                />
                <circle
                  cx="9"
                  cy="7"
                  r="4"
                />
                <path
                  d="M22 21v-2a4 4 0 0 0-3-3.87"
                />
                <path
                  d="M16 3.13a4 4 0 0 1 0 7.75"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold"
            >
              Friend Connections
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Connect with friends and build your network. Share your thoughts with people who matter to you.
            </p>
          </div>
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="mb-4 flex h-12 w-12 items-center justify-center rounded-full bg-primary/10"
            >
              <svg
                class="lucide lucide-zap h-6 w-6 text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M4 14a1 1 0 0 1-.78-1.63l9.9-10.2a.5.5 0 0 1 .86.46l-1.92 6.02A1 1 0 0 0 13 10h7a1 1 0 0 1 .78 1.63l-9.9 10.2a.5.5 0 0 1-.86-.46l1.92-6.02A1 1 0 0 0 11 14z"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold"
            >
              Smart Recommendations
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Discover content tailored to your interests based on your interactions, likes, and shares.
            </p>
          </div>
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="mb-4 flex h-12 w-12 items-center justify-center rounded-full bg-primary/10"
            >
              <svg
                class="lucide lucide-message-square h-6 w-6 text-primary"
                fill="none"
                height="24"
                stroke="currentColor"
                stroke-linecap="round"
                stroke-linejoin="round"
                stroke-width="2"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 15a2 2 0 0 1-2 2H7l-4 4V5a2 2 0 0 1 2-2h14a2 2 0 0 1 2 2z"
                />
              </svg>
            </div>
            <h3
              class="text-xl font-bold"
            >
              Engaging Discussions
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Participate in meaningful conversations through comments and replies. Build communities around shared interests.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="max-w-full overflow-clip section py-20"
      id="how-it-works"
    >
      <div
        class="container"
      >
        <div
          class="text-center mb-16"
        >
          <h2
            class="text-3xl font-bold tracking-tight sm:text-4xl md:text-5xl"
          >
            How PEPP Works
          </h2>
          <p
            class="mx-auto mt-4 max-w-2xl text-muted-foreground"
          >
            Getting started is easy. Create an account, customize your profile, and start sharing your perspective.
          </p>
        </div>
        <div
          class="grid gap-12 md:grid-cols-3 horizontal-move-from-right"
        >
          <div
            class="flex flex-col items-center text-center"
          >
            <div
              class="flex h-16 w-16 items-center justify-center rounded-full bg-primary text-primary-foreground text-2xl font-bold"
            >
              1
            </div>
            <h3
              class="mt-6 text-xl font-bold"
            >
              Create Your Account
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Sign up in seconds and set up your profile with your interests and a profile picture.
            </p>
          </div>
          <div
            class="flex flex-col items-center text-center"
          >
            <div
              class="flex h-16 w-16 items-center justify-center rounded-full bg-primary text-primary-foreground text-2xl font-bold"
            >
              2
            </div>
            <h3
              class="mt-6 text-xl font-bold"
            >
              Connect With Friends
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Find and add friends to your network. Discover people with similar interests.
            </p>
          </div>
          <div
            class="flex flex-col items-center text-center"
          >
            <div
              class="flex h-16 w-16 items-center justify-center rounded-full bg-primary text-primary-foreground text-2xl font-bold"
            >
              3
            </div>
            <h3
              class="mt-6 text-xl font-bold"
            >
              Share Your Perspective
            </h3>
            <p
              class="mt-2 text-muted-foreground"
            >
              Create posts using our wide-angle editor and engage with content from your network.
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="section py-20 bg-muted/50"
      id="testimonials"
    >
      <div
        class="container"
      >
        <div
          class="text-center mb-16"
        >
          <h2
            class="text-3xl font-bold tracking-tight sm:text-4xl md:text-5xl"
          >
            What Our Users Say
          </h2>
          <p
            class="mx-auto mt-4 max-w-2xl text-muted-foreground"
          >
            Join thousands of satisfied users who have transformed how they connect online.
          </p>
        </div>
        <div
          class="grid gap-8 md:grid-cols-3"
        >
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="flex items-center gap-4 mb-4"
            >
              <div
                class="h-12 w-12 rounded-full bg-primary/20 flex items-center justify-center"
              >
                <span
                  class="text-primary font-bold"
                >
                  JD
                </span>
              </div>
              <div>
                <h4
                  class="font-bold"
                >
                  Jane Doe
                </h4>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Content Creator
                </p>
              </div>
            </div>
            <p
              class="italic"
            >
              "The wide-angle editor has completely changed how I share my content. I can now express my ideas fully without feeling constrained."
            </p>
          </div>
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="flex items-center gap-4 mb-4"
            >
              <div
                class="h-12 w-12 rounded-full bg-primary/20 flex items-center justify-center"
              >
                <span
                  class="text-primary font-bold"
                >
                  JS
                </span>
              </div>
              <div>
                <h4
                  class="font-bold"
                >
                  John Smith
                </h4>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Photographer
                </p>
              </div>
            </div>
            <p
              class="italic"
            >
              "The recommendation system is spot on! I've discovered so many like-minded creators that I would have never found otherwise."
            </p>
          </div>
          <div
            class="bg-background rounded-lg p-6 shadow-sm border"
          >
            <div
              class="flex items-center gap-4 mb-4"
            >
              <div
                class="h-12 w-12 rounded-full bg-primary/20 flex items-center justify-center"
              >
                <span
                  class="text-primary font-bold"
                >
                  AL
                </span>
              </div>
              <div>
                <h4
                  class="font-bold"
                >
                  Alex Lee
                </h4>
                <p
                  class="text-sm text-muted-foreground"
                >
                  Student
                </p>
              </div>
            </div>
            <p
              class="italic"
            >
              "I love how easy it is to engage in meaningful discussions. The comment and reply system makes conversations flow naturally."
            </p>
          </div>
        </div>
      </div>
    </section>
    <section
      class="section py-20 bg-primary text-primary-foreground"
    >
      <div
        class="container text-center"
      >
        <h2
          class="text-3xl font-bold tracking-tight sm:text-4xl md:text-5xl"
        >
          Ready to share your perspective?
        </h2>
        <p
          class="mx-auto mt-4 max-w-2xl opacity-90"
        >
          Join our community today and start connecting with people who share your interests.
        </p>
        <div
          class="mt-10"
        >
          <a
            class="inline-flex items-center justify-center whitespace-nowrap text-sm font-medium ring-offset-background transition-colors focus-visible:outline-none focus-visible:ring-2 focus-visible:ring-ring focus-visible:ring-offset-2 disabled:pointer-events-none disabled:opacity-50 bg-secondary text-secondary-foreground hover:bg-secondary/80 h-11 rounded-md px-8 gap-2"
            href="/sign-up"
          >
            Get Started Now 
            <svg
              class="lucide lucide-arrow-right h-4 w-4"
              fill="none"
              height="24"
              stroke="currentColor"
              stroke-linecap="round"
              stroke-linejoin="round"
              stroke-width="2"
              viewBox="0 0 24 24"
              width="24"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M5 12h14"
              />
              <path
                d="m12 5 7 7-7 7"
              />
            </svg>
          </a>
        </div>
      </div>
    </section>
  </div>
</div>
`;
